seatunnel-api/src/main/java/org/apache/seatunnel/api/common/SeaTunnelPluginLifeCycle.java

```diff
@@ -31,6 +35,9 @@ public interface SeaTunnelPluginLifeCycle {
      * @param pluginConfig plugin config.
      * @throws PrepareFailException if plugin prepare failed, the {@link PrepareFailException} will
      *     throw.
+     * @deprecated SeaTunnel will not invoke prepare when init plugin, instead by {@link
+     *     org.apache.seatunnel.api.table.factory.Factory}
      */
+    @Deprecated
```

Comment:
Done. But https://github.com/apache/seatunnel/blob/dev/seatunnel-api/src/main/java/org/apache/seatunnel/api/sink/SeaTunnelSink.java#L65 doesn't deprecated. Because we don't have a replacement yet

Reply:
But it will be deprecated in the future